src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ValidationAttributeProvider.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewFeatures.Internal;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures
+{
+    /// <summary>
+    /// Default implementation of <see cref="IValidationAttributeProvider"/>.
+    /// </summary>
+    public class ValidationAttributeProvider : IValidationAttributeProvider
+    {
+        private readonly IModelMetadataProvider _metadataProvider;
+        private readonly ClientValidatorCache _clientValidatorCache;
+        private readonly IClientModelValidatorProvider _clientModelValidatorProvider;
+
+        /// <summary>
+        /// Initializes a new <see cref="ValidationAttributeProvider"/> instance.
+        /// </summary>
+        /// <param name="optionsAccessor">The accessor for <see cref="MvcViewOptions"/>.</param>
+        /// <param name="metadataProvider">The <see cref="IModelMetadataProvider"/>.</param>
+        /// <param name="clientValidatorCache">The <see cref="ClientValidatorCache"/> that provides
+        /// a list of <see cref="IClientModelValidator"/>s.</param>
+        public ValidationAttributeProvider(
+            IOptions<MvcViewOptions> optionsAccessor,
+            IModelMetadataProvider metadataProvider,
+            ClientValidatorCache clientValidatorCache)
```

Comment:
Ideally all of the features of `ClientValidatorCache` would just be part of this class, but unfortunately that shows up in API definitions, so we'd best leave it alone.